hw1/02-palindrome.js

```diff
@@ -1,3 +1,39 @@
-const elem = document.querySelector('input');
+const elem = document.querySelector("input");
+const resultDiv = document.querySelector(".result");
 
-elem.addEventListener('input', handleInput);
+elem.addEventListener("input", handleInput);
+
+function palindromeAlgorithm(userInput) {
+  let originalNumber = userInput;
+  let reversedNumber = 0;
+
+  while (userInput > 0) {
+    reversedNumber = reversedNumber * 10 + (userInput % 10);
+    userInput = Math.floor(userInput / 10);
+  }
+
+  return originalNumber === reversedNumber;
+}
+
+function isPalindrome(userInput) {
+  if (typeof userInput != "number") return false;
+  if (userInput < 0) return false;
```

Comment:
I appreciate the logic to check if the input is not a number, I hadn't thought about that edge case.
Just a small suggestion, if an error message is included in case a negative value or any sort of wrong value is input, I think it'll be easier from the user readability point of view. :)

Reply:
Okay, I implemented those suggested changes, I tested it out and it seems like letters can't be typed since the input is of type number. The only letter that can be typed is 'e' which is for an exponent -> 2e3. I also tried to get the negative number message, but I was never able to input a negative number that got that error message.